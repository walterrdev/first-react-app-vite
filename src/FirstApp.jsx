// Importar PropTypes: Sirven para establecer restricciones de uso a nuestro componente. Ej: El tipo (string, number, etc) de la propiedad.
import PropTypes from 'prop-types';

/**
 * Componente funcional FirstApp
 * 
 * Se encarga de pintar en la vista el texto (Título y subtitulo) enviado desde el componente.
 * 
 * @param {String} title | Título a mostrar en la vista.
 * @param {String} subtitulo | Subtitulo a mostrar en la vista.
 * @param {String} paragram | Párrafo a mostrar en la vista.
 * @returns 
 */
export const FirstApp = ({ title, subTitle, paragram = 'El párrafo está vacío, por favor envíe uno al componente.' }) => {

    return (
        <>
            <h1>{ title }</h1>
            <h4>{ subTitle }</h4>
            <p>{ paragram }</p>
        </>
    );
}

/**
 * Establecer restricciones de uso para el componente.
 * 
 * Todas las propiedades deben de ser de tipo String.
 * Las propiedades title y paragram son obligatorias.
 * El subTitle es opcional y tiene valor por dejecto.
 */
FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string,
    paragram: PropTypes.string.isRequired,
}

