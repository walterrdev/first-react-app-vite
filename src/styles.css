html, body {
    background-color: #21232A;
    color: white;
    font-family: Helvetica, Arial, sans-serif;
    font-size: 1.3rem;
    padding: 40px;    
}
p {
    font-size: 1rem;
}