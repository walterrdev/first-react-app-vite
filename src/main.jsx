// Importar componentes
import React from 'react';
import ReactDOM from 'react-dom/client';
import { FirstApp } from './FirstApp';

// Cargar estilos CSS a nivel global
import './styles.css';

// Renderizar componente en la vista (DOM)
ReactDOM.createRoot( document.getElementById('root') ).render(
    <React.StrictMode>
        <FirstApp title="!Hello world! I'm a awesome title" subTitle="I'm a awesome subtitle" paragram="I'm awesome paragram." />
    </React.StrictMode>
);
