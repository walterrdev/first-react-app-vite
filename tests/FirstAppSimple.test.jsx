import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Pruebas de <FirstApp />', () => { 
    const title = 'Hello world';
    const subTitle = "I'm a awesome subtitle";
    const paragram = "I'm awesome paragram.";

    test('Debe hacer match con el snapshot', () => { 
        const { container } = render ( <FirstApp title={ title } subTitle={ subTitle } paragram={ paragram } /> );
        expect( container ).toMatchSnapshot();
    });

    test('Debe mostrar título', () => {
        render ( <FirstApp title={ title } subTitle={ subTitle } paragram={ paragram } /> );
        // screen.debug();
        expect( screen.getByText(title) ).toBeTruthy();
    });

    test('Debe mostrar título en un h1', () => {
        render ( <FirstApp title={ title } subTitle={ subTitle } paragram={ paragram } /> );
        expect( screen.getByRole('heading', { level: 1 }).innerHTML ).toContain( title );
    });

    test('Debe mostrar el subtitulo enviado por props', () => {
        render ( <FirstApp title={ title } subTitle={ subTitle } paragram={ paragram } /> );
        expect( screen.getByText(subTitle) ).toBeTruthy();
    });
});
