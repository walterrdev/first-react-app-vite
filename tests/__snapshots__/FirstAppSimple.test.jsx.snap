// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <FirstApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hello world
  </h1>
  <h4>
    I'm a awesome subtitle
  </h4>
  <p>
    I'm awesome paragram.
  </p>
</div>
`;
